spec-dtslint/operators/share-spec.ts

```diff
@@ -8,3 +10,13 @@ it('should infer correctly', () => {
 it('should enforce types', () => {
   const o = of('foo', 'bar', 'baz').pipe(share('abc')); // $ExpectError
 });
+
+it('should accept observable input as notifier values', () => {
+  const o = of('foo', 'bar', 'baz').pipe( // $ExpectType Observable<string>
+    share({
+      resetOnError: () => new Promise(() => {}),
+      resetOnComplete: () => new Promise(() => {}),
+      resetOnRefCountZero: () => new Promise(() => {}),
```

Comment:
Can we use something better than unresolved forever promises here? haha. I mean, I know it's just tests, but this is a leak.

Reply:
Will change

Reply:
Out of curiosity: is it indeed a leak? No references are being held to anything?